aws/rs_aws_ec2/README.md

```diff
@@ -0,0 +1,130 @@
+1. In the Design section, use the `Upload CAT` interface to complete the following:
+   1. Upload each of packages listed in the Requirements Section
+   1. Upload the `aws_vpc_plugin.rb` file located in this repository
+ 
+## How to Use
+The VPC Plugin has been packaged as `plugin/rs_aws_vpc`. In order to use this plugin you must import this plugin into a CAT.
+```
+import "plugin/rs_aws_vpc"
+```
+For more information on using packages, please refer to the RightScale online documenataion. [Importing a Package](http://docs.rightscale.com/ss/guides/ss_packaging_cats.html#importing-a-package)
+
+AWS VPC resources can now be created by specifying a resource declaration with the desired fields. See the Supported Actions section for a full list of supported actions.
+The resulting resrouce can be manipulated just like the native RightScale resources in RCL and CAT. See the Examples Section for more examples and complete CAT's.
+## Supported Resources
+ - vpc
+ - endpoint
+ - route_table
+
+## Usage
+```
+#Creates an VPC
+resource "my_vpc", type: "rs_aws_vpc.vpc" do
+  cidr_block "10.0.0.0/16"
+  instance_tenancy "default"
+end
+
+resource "my_vpc_endpoint", type: "rs_aws_vpc.endpoint" do
+  vpc_id @my_vpc.vpcId
+  service_name "com.amazonaws.us-east-1.s3"
+end
+
+resource "my_rs_vpc", type: "rs_cm.network" do
+  name "my_rs_vpc"
+  cidr_block "10.0.0.0/16"
+  cloud_href "/api/clouds/1"
+end
+
+resource "my_rs_vpc_endpoint", type: "rs_aws_vpc.endpoint" do
+  vpc_id @my_rs_vpc.resource_uid
+  service_name "com.amazonaws.us-east-1.s3"
+end
+```
+#
+## Resources
+## vpc
+#### Supported Fields
+| Field Name | Required? | Description |
+|------------|-----------|-------------|
+|amazon_provided_ipv6_cidr_block|No|Requests an Amazon-provided IPv6 CIDR block with a /56 prefix length for the VPC. You cannot specify the range of IP addresses, or the size of the CIDR block.|
+|cidr_block|Yes|The IPv4 network range for the VPC, in CIDR notation. For example, 10.0.0.0/16.|
+|instance_tenancy|No|The tenancy options for instances launched into the VPC. For default, instances are launched with shared tenancy by default. You can launch instances with any tenancy into a shared tenancy VPC. For dedicated, instances are launched as dedicated tenancy instances by default. You can only launch instances with a tenancy of dedicated or host into a dedicated tenancy VPC.|
+
+## Supported Actions
+| Action | API Implementation | Support Level |
+|--------------|:----:|:-------------:|
+| create | [CreateVpc](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpc.html) | Supported |
+| destroy | [DeleteVpc](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DeleteVpc.html) | Supported |
+| list | [DescribeVpcs](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcs.html) | Supported |
+| routeTables | [DescribeRouteTables](http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeRouteTables.html) | Supported |
```

Comment:
add a note to "routeTables" so that people understand that it behaves more like a link